From b1edfbcf2341820760cc65b8ef59a77e769ca3f5 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Thu, 8 Feb 2024 18:12:33 +0100
Subject: [PATCH 0559/1695] dts: stm32: add the support of the M33 remote
 processor on stm32mp257 boards

Enable the m33_rproc node to support the Cortex-M33. The configuration is
the open configuration, which means that OP-TEE does not load the
Cortex-M33 firmware. The access rights for loading the remote processor
firmware are provided to the non-secure context.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c27741a9521eb0d322f130341371374e8fe6de03 dts: stm32: add the support of the M33 remote processor on stm32mp257 boards]
Change-Id: I4ee6b5ace1a8b2f832c713d156e24a9cb8fcde82
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/358619
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/dts/stm32mp257f-dk.dts  | 4 ++++
 core/arch/arm/dts/stm32mp257f-ev1.dts | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index a6b68d501..019d9832a 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -301,6 +301,10 @@
 	status = "okay";
 };
 
+&m33_rproc {
+	status = "okay";
+};
+
 &pwr {
 	status = "okay";
 
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index d54b3481c..74922e5a7 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -323,6 +323,10 @@
 	status = "okay";
 };
 
+&m33_rproc {
+	status = "okay";
+};
+
 &ommanager {
 	memory-region = <&mm_ospi1>;
 	memory-region-names = "mm_ospi1";
-- 
2.47.2

